test/jdk/jdk/modules/scenarios/automaticmodules/RunWithAutomaticModules.java

```diff
@@ -183,4 +183,65 @@ public void testServiceProvider() throws Exception {
 
     }
 
+    /**
```

Comment:
The update to ModulePatcher is fine,  that was an oversight that was missed because it's a real corner to have an automatic module be the initial module and patch it to add additional packages at the same time.

The tests for --patch-module are in jdk/tools/launcher/modules/patch. I was planning to add tests there for patching automatic modules when I created the issue (I didn't know you were going to create a PR for it).

Reply:
Choosing the right place for the test was one of the harder parts.

I did choose automaticmodules because it already has most of the necessary machinery already in place.  
But I will move the tests to jdk/tools/launcher/modules/patch.